packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -488,13 +472,22 @@ class TimeInput extends React.Component {
     } = this.state;
     const variant = TimeUtil.getVariantFromLocale(this.props);
 
+    if (event.currentTarget.value.length === 2 && event.key.match(/^[0-9]/g)) {
+      event.currentTarget.value = ''; // eslint-disable-line no-param-reassign
+    }
+
+    if (!event.key.match(/^[0-9]/g) && !(event.keyCode === KeyCode.KEY_BACK_SPACE || event.keyCode === KeyCode.KEY_DELETE || event.keyCode === KeyCode.KEY_TAB || event.keyCode === KeyCode.KEY_RIGHT || event.keyCode === KeyCode.KEY_LEFT)) {
```

Comment:
should we not include Up/Down as well here ?

Reply:
No need, because automatically Up/Down key will be increase/decrease value when we use type as number in input field